### PR TITLE
add updates blogpost

### PR DESCRIPTION
**Pull request by DhairyaLGandhi**
This still needs some links to PRs and performance comparison charts where applicable but starting off with a blogpost about the updates to this version.

### CODE REVIEW

File: blog/_posts/2021-01-04-Fluxv12.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+Zygote and IRTools have been updated to work with Julia 1.6 out of the box (shoutout to Simeon!).
+This is available with Zygote v0.6 onwards.
+
+### Some Other updates:
+* Zygote is made PackageCompiler friendly to be able to compile projects using it into system images to get rid of compilation overhead.
+* New Optimisers - OADAM, AdaBelief
+* Added label smoothing so it can be applied to supported loss functions
+* Zygote supports parallelism via `pmap`
+* OneHot{Vector/ Matrix} has been generalized as a `OneHotArray` and fixes the case of scalar indexing with GPUs in certain kernels.
+* Experimentally - `outdims` (renamed to `outsize`) uses a custom `Nil` type to help with performance and correctness issues and be more general.
```
Comment (DhairyaLGandhi) [inline blog/_posts/2021-01-04-Fluxv12.md:32]:
```suggestion
* Experimentally - `outdims` (renamed to `outputsize`) uses a custom `Nil` type to help with performance and correctness issues and be more general.
```

File: blog/_posts/2021-01-04-Fluxv12.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+* NNlib `\nablasoftmax`, `\nablalogsoftmax` optionally accept a third argument of the output of `softmax` to avoid repeating the forward pass.
+* train/testmode is now generalized to all Functors
+* Added `Flux.skip` to optionally not optimise when conditions specified in a callback are met.
+* ADAM is made more efficient by removing a large number of allocations.
+* Added `kaiming_uniform/normal` initialization options.
```
Comment (DhairyaLGandhi) [inline blog/_posts/2021-01-04-Fluxv12.md:37]:
```suggestion
* Added `kaiming_uniform/normal` initialization options.
* Added `sparse` initialisation option
```

File: blog/_posts/2021-01-04-Fluxv12.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+* Zygote supports parallelism via `pmap`
+* OneHot{Vector/ Matrix} has been generalized as a `OneHotArray` and fixes the case of scalar indexing with GPUs in certain kernels.
+* Experimentally - `outdims` (renamed to `outputsize`) uses a custom `Nil` type to help with performance and correctness issues and be more general.
+* NNlib `\nablasoftmax`, `\nablalogsoftmax` optionally accept a third argument of the output of `softmax` to avoid repeating the forward pass.
+* train/testmode is now generalized to all Functors
+* Added `Flux.skip` to optionally not optimise when conditions specified in a callback are met.
+* ADAM is made more efficient by removing a large number of allocations.
+* Added `kaiming_uniform/normal` initialization options.
+* Added `sparse` initialisation option
+
```
Comment (darsnack) [inline blog/_posts/2021-01-04-Fluxv12.md:39]:
```suggestion
* Added `Parallel` layer to represent high-level layers with multiple branches (e.g. [inception modules](https://arxiv.org/pdf/1409.4842v1.pdf))

```

File: blog/_posts/2021-01-04-Fluxv12.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+The new Flux v0.12 is out and available with a number of new features.
+
+### CUDNN and CUDA changes
+
+Some big changes the GPU stack mean that we should see a more complete coverage of Nvidia's CUDNN interface. This allows of unlocking some higher performant kernels as well to be used by default. The `softmax` function which can be compute heavy in some cases should now see faster runtime performance. There has also been speedups made to several other kernels in the CUDA side to have up to 3x faster ResNet style convolutional models.
+
+### RNNs in Julia
+
+The new RNN interface now relies on Julia code all the way through rather than using the CUDNN kernels. This gives us more flexiblity in the design moving on while being faster than the CUDNN implementation.
```
Comment (DhairyaLGandhi) [inline blog/_posts/2021-01-04-Fluxv12.md:15]:
```suggestion
```

File: blog/_posts/2021-01-04-Fluxv12.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+layout: blog
+---
+
+The new Flux v0.12 is out and available with a number of new features.
+
+### CUDNN and CUDA changes
+
+Some big changes the GPU stack mean that we should see a more complete coverage of Nvidia's CUDNN interface. This allows of unlocking some higher performant kernels as well to be used by default. The `softmax` function which can be compute heavy in some cases should now see faster runtime performance. There has also been speedups made to several other kernels in the CUDA side to have up to 3x faster ResNet style convolutional models.
+
+### RNNs in Julia
```
Comment (DhairyaLGandhi) [inline blog/_posts/2021-01-04-Fluxv12.md:13]:
```suggestion
```

**Comment by logankilpatrick** [comment]
Bumping this @DhairyaLGandhi

**Comment by logankilpatrick** [comment]
Bumping this again @DhairyaLGandhi (sorry)

**Comment by CarloLucibello** [comment]
closing as stale

**Pull request closed without merging**